### ISSUE TITLE
Update qc report process to export separate cell types report

### ISSUE THREAD

**Issue by sjspielman** [AlexsLemonade/scpca-nf#482]
_Broken out of #447_

> We can render this file within the same script that we use to render the full QC report. We just need to modify the process to include two output html files.

We will need to update the process that generates a QC report to also generate the supplemental cell type QC report if cell types are present.

**Comment by sjspielman** [comment]
Note to also include Rmd parameter value for the supplemental QC report!

**Comment by sjspielman** [comment]
There are a couple spots we need to discuss before implementing this:

- Should this report be exported from the `sce_qc_report.R` script, or should we call `Rscript -e rmarkdown::render(...)` directly from the nextflow process?
- If we export from `sce_qc_report.R`, we could figure out if cell types are present by reading in the processed SCE and checking its contents.
- Alternatively, if we export directly `Rscript -e ...`, we'll need to get indicators into the workflow for whether cell types are present:
  - The supplementary report should get rendered if _at least one_ of the following is present: `submitter`, `singler`, `cellassign`. 
  - Perhaps we revive using indicator variables like `meta.has_<ONE OF THOSE>`  into the workflow? For cell typing methods, we'd define these when the `classify_<singler or cellassign>` process is called. For submitter annotations, we'd add it [here](https://github.com/AlexsLemonade/scpca-nf/blob/3bf3e7186efbf866fc5ef2816ee39a167fd891b3/modules/sce-processing.nf#L13-L39) (add check for submitter file and assign meta variable before entering the bash parts), and similarly into `make_unmerged_filtered_sce`; that said, we could actually define the submitter indicator anytime we feel like checking that `submitter_cell_types_file` exists. This means we don't actually need a variable; we could just check the submitter file directly in the QC process when we check for `meta.has_singler` & `meta.has_cellassign`.

**Comment by jashapiro** [comment]
I would favor adding this to the  `sce_qc_report.R` script, triggered by an option to specify the cell annotation report file path.

We should not need additional indicator variable(s), the presence of a submitter file or annotation reference/model file should be sufficient. Note again that testing for "truthiness" of those variables can act exactly like an indicator variable: The lack of a file path will be treated as false by nextflow/groovy in boolean context.
So something like:

```
has_celltype = params.perform_celltyping & (meta.submitter_cell_types_file | meta.singler_model_file | meta.cellassign_reference_file )
celltype_report_path = "${meta.library_id}_celltype-report.html"
"""
...
sce_qc_report.R \
...
${has_celltype ? "--cell_type_report_file ${celltype_report_path}" : ''} \
...
"""
```

**Comment by sjspielman** [comment]
One further point of discussion:

The `scpcaTools` function seems to be heavily tailored (reasonably!) to the main QC report, e.g., checks for unfiltered and filtered SCEs, which aren't needed for the cell type report. In addition, the packages that it checks for (https://github.com/AlexsLemonade/scpcaTools/blob/139a001cd590dc85df25db39597cd3d26c48c047/R/generate_qc_report.R#L26-L43) don't include `ggridges` and `ComplexHeatmap`, which the cell type report requires.

So, we'll either need a new `scpcaTools` function for this report, or we can directly write out the `rmarkdown::render()` code in `sce_qc_report.R` (or a function could live in `scpca-nf`, but if we're writing a function I think it should be in `scpcaTools` for organizational reasons).

That said, I wonder if those package installation checks are really necessary given that they are all expected to be in the docker container that this process uses? They don't hurt though!

Noting I'll bring this up at our pre-planning meeting soon!

**Comment by jashapiro** [comment]
I would skip the `scpcaTools` function and simply use the `rmarkdown::render()`. Putting the QC report in `scpcaTools` may have been a mistake!

**Comment by sjspielman** [comment]
Closed by #496
